io.helidon.messaging.connectors.jms.AckMpTest randomly fails

I saw this test failure several times, restarting the job usually helps.
```
[INFO] Running io.helidon.messaging.connectors.jms.AckMpTest
2023.06.15 09:57:15 INFO io.helidon.microprofile.cdi.HelidonContainerImpl Thread[#22,helidon-cdi-shutdown-hook,5,main]: Shutdown requested by JVM shutting down
2023.06.15 09:57:15 INFO io.helidon.nima.webserver.ServerListener Thread[#27,server-@default-listener,5,main]: [0x75af79bc] @default socket closed.
2023.06.15 09:57:15 INFO io.helidon.nima.webserver.LoomServer Thread[#22,helidon-cdi-shutdown-hook,5,main]: Níma server stopped all channels.
2023.06.15 09:57:15 INFO io.helidon.microprofile.server.ServerCdiExtension Thread[#22,helidon-cdi-shutdown-hook,5,main]: Server stopped in 8 milliseconds.
2023.06.15 09:57:15 INFO io.helidon.microprofile.cdi.HelidonContainerImpl Thread[#22,helidon-cdi-shutdown-hook,5,main]: Shutdown finished
[INFO] Async process complete, exit value = 0 for: [java, --enable-preview, -jar, /tmp/maven-exec15510170939562921931.jar, -classpath]
Error:  Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6.693 s <<< FAILURE! - in io.helidon.messaging.connectors.jms.AckMpTest
Error:  io.helidon.messaging.connectors.jms.AckMpTest.resendAckTestPart2(SeContainer)  Time elapsed: 5.185 s  <<< FAILURE!
java.lang.AssertionError: 
Not all 3 items delivered in time.
Expected: iterable containing ["NO_ACK-1", "NO_ACK-2", "NO_ACK-3"]
     but: item 0: was "NO_ACK-2"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at io.helidon.messaging.connectors.mock.MockOutgoing.awaitData(MockOutgoing.java:166)
	at io.helidon.messaging.connectors.mock.MockOutgoing.awaitPayloads(MockOutgoing.java:194)
	at io.helidon.messaging.connectors.jms.AckMpTest.resendAckTestPart2(AckMpTest.java:116)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.api.extension.InvocationInterceptor.interceptTestMethod(InvocationInterceptor.java:117)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:188)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
```
Last time I saw it in PR #6959
https://github.com/helidon-io/helidon/actions/runs/5277095950/jobs/9544670221?pr=6959

Tested locally on the same branch (from PR) and test passed.

--- a/tests/integration/jms/src/test/java/io/helidon/messaging/connectors/jms/AbstractMPTest.java
+++ b/tests/integration/jms/src/test/java/io/helidon/messaging/connectors/jms/AbstractMPTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -49,7 +49,7 @@ public abstract class AbstractMPTest extends AbstractJmsTest {
                                    final List<String> expected, Consumer<TextMessage> messageConsumer) {
         consumingBean.expectedRequests(expected.size());
         produce(topic, testData, messageConsumer);
-        if (expected.size() > 0) {
+        if (!expected.isEmpty()) {
             // Wait till records are delivered
             boolean done = consumingBean.await();
             assertThat(String.format("Timeout waiting for results.\nExpected: %s \nBut was: %s",
@@ -65,11 +65,12 @@ public abstract class AbstractMPTest extends AbstractJmsTest {
             Destination dest = topic.startsWith("topic")
                     ? session.createTopic(topic)
                     : session.createQueue(topic);
-            MessageProducer producer = session.createProducer(dest);
-            for (String s : testData) {
-                TextMessage textMessage = session.createTextMessage(s);
-                messageConsumer.accept(textMessage);
-                producer.send(textMessage);
+            try (MessageProducer producer = session.createProducer(dest)) {
+                for (String s : testData) {
+                    TextMessage textMessage = session.createTextMessage(s);
+                    messageConsumer.accept(textMessage);
+                    producer.send(textMessage);
+                }
             }
         } catch (JMSException e) {
             throw new RuntimeException(e);
@@ -81,17 +82,17 @@ public abstract class AbstractMPTest extends AbstractJmsTest {
             Destination dest = topic.startsWith("topic")
                     ? session.createTopic(topic)
                     : session.createQueue(topic);
-            MessageConsumer consumer = session.createConsumer(dest);
-            Message m;
             List<Message> result = new ArrayList<>();
-            for (; ; ) {
-                m = consumer.receive(500L);
-                if (m == null) {
-                    break;
+            try (MessageConsumer consumer = session.createConsumer(dest)) {
+                Message m;
+                for (; ; ) {
+                    m = consumer.receive(500L);
+                    if (m == null) {
+                        break;
+                    }
+                    result.add(m);
                 }
-                result.add(m);
             }
-            consumer.close();
             return result.stream();
         } catch (JMSException e) {
             throw new RuntimeException(e);

--- a/tests/integration/jms/src/test/java/io/helidon/messaging/connectors/jms/AckMpTest.java
+++ b/tests/integration/jms/src/test/java/io/helidon/messaging/connectors/jms/AckMpTest.java
@@ -24,13 +24,13 @@ import io.helidon.messaging.connectors.mock.MockConnector;
 import io.helidon.messaging.connectors.mock.TestConnector;
 import io.helidon.microprofile.config.ConfigCdiExtension;
 import io.helidon.microprofile.messaging.MessagingCdiExtension;
-import io.helidon.microprofile.testing.junit5.AddBean;
-import io.helidon.microprofile.testing.junit5.AddBeans;
-import io.helidon.microprofile.testing.junit5.AddConfig;
-import io.helidon.microprofile.testing.junit5.AddConfigs;
-import io.helidon.microprofile.testing.junit5.AddExtension;
-import io.helidon.microprofile.testing.junit5.AddExtensions;
-import io.helidon.microprofile.testing.junit5.DisableDiscovery;
+import io.helidon.microprofile.testing.AddBean;
+import io.helidon.microprofile.testing.AddBeans;
+import io.helidon.microprofile.testing.AddConfig;
+import io.helidon.microprofile.testing.AddConfigs;
+import io.helidon.microprofile.testing.AddExtension;
+import io.helidon.microprofile.testing.AddExtensions;
+import io.helidon.microprofile.testing.DisableDiscovery;
 import io.helidon.microprofile.testing.junit5.HelidonTest;
 
 import jakarta.enterprise.inject.se.SeContainer;
@@ -43,8 +43,6 @@ import org.junit.jupiter.api.MethodOrderer;
 import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestMethodOrder;
-import org.junit.jupiter.api.condition.DisabledOnOs;
-import org.junit.jupiter.api.condition.OS;
 
 import static java.lang.System.Logger.Level.DEBUG;
 
@@ -77,6 +75,7 @@ import static java.lang.System.Logger.Level.DEBUG;
 public class AckMpTest extends AbstractMPTest {
 
     static final String TEST_QUEUE_ACK = "queue-ack";
+    static final Duration TIMEOUT = Duration.ofSeconds(15);
 
     private static final System.Logger LOGGER = System.getLogger(AckMpTest.class.getName());
     private static final Annotation TEST_CONNECTOR_ANNOTATION = MockConnector.class.getAnnotation(TestConnector.class);
@@ -99,24 +98,21 @@ public class AckMpTest extends AbstractMPTest {
     @Order(1)
     void resendAckTestPart1(SeContainer cdi) {
         MockConnector mockConnector = cdi.select(MockConnector.class, TEST_CONNECTOR_ANNOTATION).get();
-        //Messages starting with NO_ACK is not acked by ChannelAck bean
+        //Messages starting with NO_ACK are not acked by ChannelAck bean
         List<String> testData = List.of("0", "1", "2", "NO_ACK-1", "NO_ACK-2", "NO_ACK-3");
         produce(TEST_QUEUE_ACK, testData, m -> {});
         mockConnector.outgoing("mock-conn-channel", String.class)
-                        .awaitPayloads(Duration.ofSeconds(5), testData.toArray(String[]::new));
+                        .awaitPayloads(TIMEOUT, testData.toArray(String[]::new));
     }
 
     @Test
     @Order(2)
-    @DisabledOnOs(value = OS.WINDOWS, disabledReason = "https://github.com/helidon-io/helidon/issues/8509")
     void resendAckTestPart2(SeContainer cdi) {
             MockConnector mockConnector = cdi.select(MockConnector.class, TEST_CONNECTOR_ANNOTATION).get();
 
             //Check if not acked messages are redelivered
             mockConnector.outgoing("mock-conn-channel", String.class)
-                    .requestMax()
-                    .awaitCount(Duration.ofSeconds(5), 1)
-                    .awaitPayloads(Duration.ofSeconds(5), "NO_ACK-1", "NO_ACK-2", "NO_ACK-3");
+                    .awaitPayloads(TIMEOUT, "NO_ACK-1", "NO_ACK-2", "NO_ACK-3");
     }
 
     @AfterAll

